src/steps/account/converters.ts

```
@@ -9,17 +9,23 @@ export function getAccountKey(id: string): string {
   return `zoom_account:${id}`;
 }
 
+// TODO We're using the data available in api/v2/users/me to create the account
+// because the Account name is only available via api/v2/accounts/me or calling
+// it directly at api/v2/accounts/<accountID> and both options reside in the 
+// 'Master Account API' area and require :account:master OAuth token permissions.
+// If this becomes available without elevated permissions in the future, we
+// should change the account name below to be set to the name instead of the
+// account_number that we have right now.
 export function createAccountEntity(data: ZoomUser): Entity {
   return createIntegrationEntity({
     entityData: {
       source: data,
       assign: {
         _class: Entities.ACCOUNT._class,
         _type: Entities.ACCOUNT._type,
-        _key: getAccountKey(data.id),
-        username: data.email,
-        name: `${data.first_name} ${data.last_name}`,
-        email: data.email,
+        _key: getAccountKey(data.account_id),
+        accountId: data.account_id,
+        name: `${data.account_number}`,
```

Nice to see we're switching to 'Account' after all :slightly_smiling_face: 

I just have one quick question (I'm guessing some discussion took place regarding this and you may know the answer) - if returned `ZoomUser` didn't have `account_id` and `account_number`, would we still switch to using the class `Account` even if we had to have `displayName`/`name` be set to e.g. user's full name or email and we had to use similar things for the `_key`?

Just wondering how to approach this _in general_.

---

Good question!  I believe most users are a member of an account, even if it is just their own for personal use.  That said, it is *not* a required field in the return, so in the case that it doesn't exist, it will fail the Account creation step.  

From walking through the data with someone from our internal security team, I believe that is actually the preferred execution results.  Having an account type that essentially looked like a duplicate of an existing user immediately caused confusion.  Better to let the account fail and report that failure for them to fix.  Users, Roles, Groups, and their associated relationships will still work, they just won't be related to an overall account.

That said, this is a great call out for an area that we should think about having better error handling and reporting in the future.  Thanks!